app/models/facet_collection.rb

```diff
@@ -20,6 +20,17 @@ def filters
     facets.select(&:filterable?)
   end
 
+  def ensure_minimum_one_open_facet(facets_to_ensure)
+    at_least_one_facet_selected = (facets_to_ensure.map { |f|
+      # TODO figure out where to move this method so that it only applies to `OptionSelectFacet`. Then can remove the `defined?` check.
+      (defined? f.close_facet?) ? !f.close_facet? : false
+    }).any?
```

Comment:
I agree with your comment, perhaps add `close_facet?` to `FilterableFacet` or `Facet` defaulting to false, or make it into a module and include it in the necessary facet classes - either way a default of false and overwriting the method where necessary seems sensible